zerver/worker/queue_processors.py

```diff
@@ -586,7 +586,7 @@ def consume(self, event: Mapping[str, Any]) -> None:
                 if event['trigger'] == 'mention':
                     message['content'] = extract_query_without_mention(
                         message=message,
-                        client=self.get_bot_api_client(user_profile),
+                        client=cast(ExternalBotHandler, self.get_bot_api_client(user_profile)),
```

Comment:
Is this the right answer, or was `EmbeddedBotHandler` supposed to be a subclass of `ExternalBotHandler`, or is something else going on?

Reply:
I think they were supposed to be parallel classes, but we've done some restructuring since then (and I haven't thought about this part of the system in a long time).  Let me see.

Reply:
I think probably ideally they'd be both subclasses of some common base class?